core-rust/state-manager/src/types.rs

```diff
@@ -513,6 +513,8 @@ impl From<Hash> for TransactionHash {
     }
 }
 
+impl IsHash for TransactionHash {}
```

Comment:
Hmh, I'm not a big fan of this naming convention (TransactionHash, ReceiptHash). TransactionHash sounds like it was "a hash of a transaction", while in reality it's "a hash of a node in the transactions merkle tree". How about `TransactionTreeHash` or `TxTreeHash` for short? or `TransactionsHash`?

Reply:
I agree, `TransactionTreeHash. ReceiptTreeHash` sound super unambiguous 👍